tools/fragpipe/macros.xml

```diff
@@ -0,0 +1,1206 @@
+                        </param>
+                        <param name="tolerance" type="integer" optional="true" label="Tolerance"/>
+                        <param name="top3_pep" type="select" optional="true" label="use top 3 most intense peptide ions as part of the reference sample abundance estimation">
+                            <option value="true">Yes</option>
+                            <option value="false">No</option>
+                        </param>
+                        <param name="print_RefInt" type="select" optional="true" label="print individual reference sample abundance estimates for each multiplex in the final reports">
+                            <option value="false">No</option>
+                            <option value="true">Yes</option>
+                        </param>
+                    </section>
+                </when>
+            </conditional>
+        </section>
+    </xml>
+
+    <!-- Set isobaric quantification workflow options -->
+    <token name="@ISOBARIC_QUANTIFICATION@"><![CDATA[
+        #set $prfx = $wf.quant_iso.isobaric_quantification
+        #if $prfx.isobaric_quantification_step == 'no'
+            #set $wfdict['tmtintegrator.run-tmtintegrator'] = 'false'
+        #elif $prfx.isobaric_quantification_step == 'yes'
+            #set $wfdict['tmtintegrator.run-tmtintegrator'] = 'true'
+            #set $wfdict['tmtintegrator.dont-run-fq-lq'] = 'false'
+            #set $cxt = $prfx.options
+            #set $wfdict['tmtintegrator.quant_level'] = $cxt.quant_level
+            #set $wfdict['tmtintegrator.channel_num'] = $cxt.channel_num
+            #if $cxt.ref_tag == 'Virtual'
+                #set $wfdict['tmtintegrator.add_Ref'] = -1
+            #else
+                #set $wfdict['tmtintegrator.add_Ref'] = 1
+                #set $wfdict['tmtintegrator.ref_tag'] = $cxt.ref_tag
+            #end if
+            #if $cxt.groupby != ''
+                #set $wfdict['tmtintegrator.groupby'] = $cxt.groupby
+            #end if
+            #set $wfdict['tmtintegrator.log2transformed'] = $cxt.log2transformed
+            #if $cxt.prot_norm != 'None'
+                #set $wfdict['tmtintegrator.prot_norm'] = $cxt.prot_norm
+            #end if
+            #if cxt.glycan_conditional.use_glycan_composition == 'yes'
+                #if $cxt.glycan_conditional.mod_tag != 'None'
+                    #set $wfdict['tmtintegrator.mod_tag'] = $cxt.glycan_conditional.mod_tag
+                #end if
+                #if $cxt.glycan_conditional.min_site_prob != ''
+                    #set $wfdict['tmtintegrator.min_site_prob'] = $cxt.glycan_conditional.min_site_prob
+                #end if
+                #if $cxt.glycan_conditional.glyco_qval != ''
+                    #set $wfdict['tmtintegrator.glyco_qval'] = $cxt.glycan_conditional.glyco_qval
+                #end if
+            #end if
+            #if $cxt.unique_gene != 'None'
```

Comment:
I'm pretty sure that unset optional selects evaluate to `None`. Needs to be checked  everywhere.

Reply:
I actually put a bit of time into getting this and the below comment (`str($cxt.min_pep_prob) != ''`) right. Here's what my tests shows. I couldn't get builtin `type()` working in the cheetah so I'm just printing the evaluation:

```
#set $unique_gene_test_str_None = $cxt.unique_gene == 'None'
#set $unique_gene_test_bool_None = $cxt.unique_gene == None
#set $min_pep_prob_test_str = $cxt.min_pep_prob == ''
unique_gene_test_str_None $unique_gene_test_str_None
unique_gene_test_bool_None $unique_gene_test_bool_None
min_pep_prob_test_str $min_pep_prob_test_str
```

Result:
```
unique_gene_test_str_None True
unique_gene_test_bool_None False
min_pep_prob_test_str True
```